src/main/java/tictactoe/Minimax.java

```diff
@@ -0,0 +1,81 @@
+package tictactoe;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Minimax {
+
+    public int chooseMove
+            (Board board, String currentPlayersMark, String opponentsMark) {
+
+        var copyBoard = board.copyBoard();
+        return findBestMove(copyBoard, 0, currentPlayersMark, opponentsMark);
+    }
+
+    private int findBestMove
+            (Board board, int depth, String currentPlayersMark, String opponentsMark) {
+        var movesAndScores = new HashMap<Integer, Integer>();
+        var availableSquares = board.availableSquares(currentPlayersMark, opponentsMark);
+
+        if (board.isComplete(currentPlayersMark, opponentsMark)) {
+            return scoreMove(board, depth, currentPlayersMark, opponentsMark);
```

Comment:
I like how you have encapsulated different functionality within separate methods. Do you think the code would be easier to read and follow if the methods are written in the order they are called?

Reply:
That's a good point. I'll think it would be easier to read, if the methods are written in the order they're called. I'll change this, thank you :)